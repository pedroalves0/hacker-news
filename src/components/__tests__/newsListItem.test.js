import React from 'react';
import { shallow } from 'enzyme';
import NewsListItem from '../newsList/newsListItem';

describe('components/newsList/newsListItem', () => {
    test('should render a story if url is passed', () => {
        const wrapper = shallow(
            <NewsListItem
                id={100}
                by={'author'}
                time={1512157581}
                title={'title'}
                url={'http://www.url.com/url/url/'}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should render an ask story if url is not passed', () => {
        const wrapper = shallow(
            <NewsListItem
                id={100}
                by={'author'}
                time={1512157581}
                title={'title'}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });
});