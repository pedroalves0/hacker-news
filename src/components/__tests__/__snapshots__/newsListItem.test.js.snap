// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/newsList/newsListItem should render a story if url is passed 1`] = `
<li
  className="newslist-item"
>
  <Story
    by="author"
    id={100}
    time={1512157581}
    title="title"
    url="http://www.url.com/url/url/"
  />
</li>
`;

exports[`components/newsList/newsListItem should render an ask story if url is not passed 1`] = `
<li
  className="newslist-item"
>
  <Ask
    by="author"
    id={100}
    time={1512157581}
    title="title"
  />
</li>
`;
